GOVERNANCE.md

```diff
@@ -22,30 +18,164 @@ New maintainers must be nominated by an existing maintainer and must be elected
+
+## When to Submit a Proposal
+If there is significant risk with a potential feature or track of work (such as complexity, cost to implement,
+product viability, etc.), then we recommend creating a proposal for feedback and approval. If a potential
+feature is well understood and doesn't impose risk, then we recommend a standard GitHub issue to clarify the details.
+
+If you are considering creating a PR to change Pinniped's source code, and you are not sure if the change is
+significant enough to require using the proposal process, then please ask the maintainers.
+
+If you would like to simply share a problem that you are having, or share an idea for a potential feature,
+and you are not planning on designing a technical solution or submitting an implementation PR, then please feel free
+to create a standard GitHub issue instead of using the proposal process.
+
+## How to Submit a Proposal
+To create a proposal, submit a PR to this repo introducing a new subdirectory under the `proposals` directory
+with a terse name (for example, `0001_my-feature-name/`) prefixed by a monotonically incrementing proposal number.
+In that new subdirectory, create a `README.md` containing the core proposal.
+Include other files as necessary to help support understanding of the feature.
+
+To make your new proposal know to all other contributors, please send a link to your new proposal PR
+on the Kubernetes Slack in [#pinniped](https://kubernetes.slack.com/archives/C01BW364RJA)
+or via the [Pinniped mailing list](mailto:project-pinniped@googlegroups.com).
+
+Author(s) of proposals for major changes will give a time period of no less than five (5) working days
+for comment and remain cognizant of popular observed world holidays.
+
+A proposal must be sponsored (or co-authored) by at least one maintainer.
+If you need to find a maintainer to sponsor your proposal, then reach out via Slack or the mailing list
+with a description of the problem statement. If one or more of the maintainers agrees that the problem
+statement is within the scope of the project (see [SCOPE.md](SCOPE.md)) and is appropriate to be addressed by a proposal,
+then a maintainer will be assigned as your proposal's sponsor. The sponsor can provide you with support during
+the drafting of the proposal, including sharing additional project and roadmap context as it relates to your
+problem statement, answering questions, giving feedback, etc.
+
+### Proposal Template
+The below template is an example `README.md` for a new proposal.
+Other than the high-level details at the top of the template (title, status, author, and sponsor),
+please use whichever sections make the most sense for your proposal.
+
+```md
 ---
-# Updating Governance
-All substantive changes in Governance require a supermajority agreement by all maintainers.
\ No newline at end of file
+title: "The Name of My Proposal"
+authors: [ "@margocrawf", "@enj" ]
+status: "draft"
+sponsor: [ "@cfryanr" ]
+---
+
+# <Proposal Title>
```

Comment:
High level items I think should be added to this doc:

- User stories (dedicated section)
- Explicit API changes (i.e. knobs, new CRs, etc)
- Implicit API changes (i.e. change in behavior)
- Upgrade and downgrade semantics
- Backwards compatibility semantics
- High level description of integration tests (including changes to CI)
- Any new dependencies added (Go, C, infra like AD, etc)
- Performance considerations
- Logging considerations (at some point in the future: metrics and audit logs)
- Alternative implementations considered

Reply:
I added all of these sections except for an explicit requirement for user stories. I added a section for "Implementation Plan" which could be used to outline user stories and to talk about who will implement the design (maintainers vs somebody else).

There are already sections for problem statements, use cases, API/behavior changes, etc. All of these things together imply user stories the same way that they imply an implementation. A community member might not be familiar with our process of writing user stories, so that might be a barrier for them. I also imagine that a proposal should be accepted before time is spent crafting detailed user stories to drive the implementation.

Reply:
@cfryanr Wondering if there is a way to define an MVP in the Proposal description/Implementation plan.

Reply:
@anjaltelang Added.